### PR TITLE
Add git extension API for branches

### PR DESCRIPTION
**Pull request by alexr00**
Enables microsoft/vscode-pull-request-github#1781

### CODE REVIEW

File: extensions/git/src/git.ts [hunk 1]
```diff
@@ -1913,6 +1917,11 @@ export class Repository {
 		}
 	}
 
+	async getBranches(query: BranchQuery): Promise<Ref[]> {
+		const refs = await this.getRefs({ contains: query.contains });
+		return query.remote ? refs : refs.filter(value => !value.remote);
```
Comment (joaomoreno) [inline extensions/git/src/git.ts:1922]: We still gotta filter out everything which is not a `Head`.

File: extensions/git/src/git.ts [hunk 2]
```diff
@@ -1913,6 +1917,11 @@ export class Repository {
 		}
 	}
 
+	async getBranches(query: BranchQuery): Promise<Ref[]> {
```
Comment (joaomoreno) [inline extensions/git/src/git.ts:1920]: You can still return `string[]` if you want. 👍

**Review by joaomoreno — APPROVED** [review]
Small comments, other than that all good!

File: extensions/git/src/git.ts (hunk 2 above)
Comment (alexr00) [reply extensions/git/src/git.ts:1920]: Might as well return refs, since they have more info. It is also better for when using the `remote` query.

**Comment by alexr00** [comment]
Thanks for reviewing!

**Pull request merged**